### ISSUE TITLE
Script dialog box is expanding after using browser backpage (FFox)

### ISSUE THREAD

**Issue by doppler56** [ccloli/E-Hentai-Downloader#251]
Grease monkey with e-hentai-downloader (9/9/2023 updated)
Firefox 118.0b6 and some earlier versions too.

Noticed the script dialog box (which is injected into the html webpage code).  Has started to expand on using the back button.  If I view a pic on a page upon using the browser back button I find 2 dialog boxes.  If I view another pic then go back.  I find 3 dialog boxes.  This will keep going until I refresh or go to another page # in the chapter I am viewing.

Seems like the script is injecting upon it's self with each browser back usage instead of only the website original page.  BTW using one of the multiple script boxes to get something not a good idea.  Strange things happen.

Thanks

**Comment by ccloli** [comment]
The latest update only changes the way to calculate estimate cost. I guess that's because the browser cached previous content, but GreaseMonkey doesn't notice that and run the script again. Try another browser or another extension may help? Seems that you're using a beta version of Firefox.

**Comment by doppler56** [comment]
Wonder why email respond to your message does not show here.  But I sent this before:
It has happened on previous non-beta versions of FF.  Been happening about a month or so now.  I will purge the cache and see what happens.

--- new---
Purging the cache did nothing.  same-o-same-o.  I don't like the other available browsers.  Firefox works with extensions I need.  I know what the work around is and I can deal with that.  Maybe the work around can give a insight into what is happening.

Fact one: loading the first new page of a gallery the script run correctly.
Fact two: Using the browser back button does not reload from the website.  Content is loaded from local cache.  Problem shows itself.
Fact three:  This is a fun fact. On the new page click a graphic and back.  The problem shows.  Click on the next page.  Script runs correctly on new page.  Click on a graphic and back.  The problem shows.  Click back.  Previous page problem expands.  Click browser forward "via mouse" (not website next page button). Next page problem expands.  Using mouse browser page forward and back.  Problem expands.
Fact four: While playing with fact three.  Clicking on the website next page.  The next page script run correctly.  Even if it was messed up before.

Final fact: Problem corrects or run correctly using the website navigation buttons.  Only fails using mouse browser forward and back.  This problem is very cache related.

If the problem is within grease monkey.  Is there another script extension I can use in fire fox.  That works with your downloader script ?

**Comment by ccloli** [comment]
The problem you said is more like the browser uses some preload or network optimize feature. Instead of requesting the server again, your browser reuse previous rendered page. Maybe you can check your browser settings or extension to see if you enabled such feature.

I don't think switch to another extension would fix the issue, but you can try [Tampermonkey](http://tampermonkey.net/) or [Violentmonkey](https://violentmonkey.github.io/) to see if it helps. Consider you don't like other browsers, I guess you can try [Violentmonkey](https://violentmonkey.github.io/) first since it's opensource.

BTW does Firefox Private Browsing help? If it works fine in private browsing, I think maybe an extension in your browser breaks it  up.

**Comment by doppler56** [comment]
Actually before reading your last response I investigated another script extension engine.  Tampermonkey.  I came back to message to report it works as expected.

Grease Monkey has not been updated in over 3 YEARS.  That was the source of the problem.  I don't have a need to change open source.  Violentmonkey is available if Tampermonkey want to play games with there code or business mode.  The fact an alternative exsits it's unlikely Tampermonkey is going play games.

The only reason I was using greasemonkey was it worked and I have been using it 5+ years.  Now that it's borked.  Going to uninstall it.  With firefox sync all my other machine home and work will too.

Closing this out and thanks for your assistance.
